src/charm_state.py

```diff
@@ -161,11 +165,13 @@ def proxy(self) -> "ProxyConfig":
             no_proxy=no_proxy,
         )
 
+    # pylint: disable=too-many-arguments
```

Comment:
Perhaps consider grouping some of these into sub classes? Or otherwise, should we include an explanatory comment as to why the disable is necessary please?

Reply:
Added a comment - this is due to the second datasource and given it might be temporary and we move this in a separate charm I'm not tempted to group these and refactor and rewrite the tests associated.